### ISSUE TITLE
3 servers, 2 nextcloud, 1 collabora

### ISSUE THREAD

**Issue by deusoz** [nextcloud/richdocuments#760]
If NextcloudServer1 user "jon" jon@nc.tld1  opens document via other server running Collabora, then and different guy named "jon" on NextcloudServer2  jon@nc.tld2 opens a document, the Collabora admin panel counts these both as 1 user in the "Users online" tab.  It says correctly in large letters, that there are 2 users, but the "Users online" Collabora Admin tab @loleaflet/dist/admin/admin.html combines any users with the same name even if completely different people.

I'm wondering if there's anyway to include the domain in the username that gets sent to collabora server or if this has another workaround, or is this a Collabora bug or perhaps there's a loolwsd admin setting I can't find.  Thanks for any suggestions.  I was thinking this would not be good for the purpose of monitoring usage for 2 NextCloud instances sharing a Collabora server.

**Comment by deusoz** [comment]
![Screenshot_2019-12-29 Collabora Online Development Edition (CODE) - Admin console](https://user-images.githubusercontent.com/3331362/71560372-59845980-2a69-11ea-822b-3d972ca9bcce.png)

**Comment by deusoz** [comment]
2 users named db5d, one from a NC in Los Angeles, other from unconnected NC in Spain, sharing a server running Collabora

**Comment by juliushaertl** [comment]
I agree that we probably should add the remote url to the username that is sent to collabora then.

**Comment by deusoz** [comment]
I don't know where to start.  Collabora stand-alone server needs to use full name instead of only username or else it's a security nightmare once you share the same stand-alone among different Nextcloud deployments.

**Comment by deusoz** [comment]
A step in the right direction here... 

line 164 lib/Controller/WopiController.php  ( I added the BOLD part after getDisplayName() and now at on the Collabora Admin Panel "Documents" list it shows who belongs to each document, albeit the "Users" list still combines them and uses the "display name" of first user.

$userDisplayName = $user !== null && !$isPublic ? $user->getDisplayName()**."@".$_SERVER['HTTP_HOST']** : $wopi->getGuestDisplayname();


![image](https://user-images.githubusercontent.com/3331362/95930511-efa22a00-0dc6-11eb-8be5-500d78b64f47.png)
